### PR TITLE
Updates p2p test codes

### PR DESCRIPTION
**Pull request by minsulee2**
Added some test codes for p2p network. This PR is for p2p (client and server) status test cases and util test cases. To avoid too long PR, already too long tho, I made partial PR beforehand. The messaging parts(mock data 2.0.0 test) are NOT included. 

Updates:
- Add test cases for the p2p/util.js
- Add test cases for p2p client(index) and server status

### CODE REVIEW

File: p2p/index.js [hunk 1]
```diff
@@ -44,7 +43,7 @@ class P2pClient {
         this, node, minProtocolVersion, maxProtocolVersion, this.maxInbound);
     this.trackerWebSocket = null;
     this.outbound = {};
-    this.startHeartbeat();
+    // this.startHeartbeat();
```
Comment (platfowner) [inline p2p/index.js:46]: Is this intended? What about adding some comments?

File: package.json [hunk 2]
```diff
@@ -51,6 +53,7 @@
     "node-cron": "^2.0.3",
     "ntpsync": "^0.2.3",
     "object-sizeof": "^1.5.1",
+    "process": "^0.11.10",
```
Comment (platfowner) [inline package.json:56]: It seems we don't need to install it explicitly?

File: p2p/util.js [hunk 3]
```diff
@@ -16,17 +15,40 @@ const {
...
+      return false;
+    });
+    if (confirmed) {
+      return true;
+    }
+    await ChainUtil.sleep(1000);
+  }
+}
+
+// FIXME(minsulee2): need discussion that this is p2p util?
```
Comment (platfowner) [inline p2p/util.js:39]: Maybe we can move these util functions to a separate file such as common/network-util.js?

File: p2p/util.js [hunk 4]
```diff
@@ -103,39 +106,65 @@ function removeSocketConnectionIfExists(connectionObj, address) {
...
-  return ainUtil.ecSignMessage(JSON.stringify(messageBody), Buffer.from(privateKey, 'hex'));
+  if (!ChainUtil.isDict(messageBody)) {
+    logger.error('The message body must be the object type.');
+    return null;
+  }
+  let privateKeyBuffer;
+  try {
+    privateKeyBuffer = Buffer.from(privateKey, 'hex');
+  } catch {
+    return null;
```
Comment (platfowner) [inline p2p/util.js:124]: logging some error message?

File: p2p/util.js [hunk 5]
```diff
@@ -103,39 +106,65 @@ function removeSocketConnectionIfExists(connectionObj, address) {
...
+    logger.error('The private key is not optional but mandatory or worng private key is typed.');
+    return null;
+  }
+  return ainUtil.ecSignMessage(JSON.stringify(messageBody), privateKeyBuffer);
 }
 
 function getAddressFromMessage(message) {
-  const hashedMessage = ainUtil.hashMessage(JSON.stringify(message.data.body));
-  return ChainUtil.getAddressFromSignature(hashedMessage, message.data.signature);
+  const body = _.get(message, 'data.body');
```
Comment (platfowner) [inline p2p/util.js:134]: Can this repeated block be extracted as a separate function like isValidMessage()?

File: integration/p2p.test.js [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+const minProtocolVersion = min === undefined ? CURRENT_PROTOCOL_VERSION : min;
+const maxProtocolVersion = max;
+
+const node = new BlockchainNode();
+
+describe("p2p", () => {
+  let p2pClient;
+  let p2pServer;
+  before(() => {
+    p2pClient = new P2pClient(node, minProtocolVersion, maxProtocolVersion);
```
Comment (platfowner) [inline integration/p2p.test.js:31]: As blockchain nodes are not actually run in these test cases (i.e., only a module is executed as part of the test job), what about moving this test file under unittest? unittest/blockchain.test.js seems to be a similar case?

File: p2p/index.js (hunk 1 above)
Comment (minsulee2) [reply p2p/index.js]: Oh, uncommented it again thanks!

File: integration/p2p.test.js [hunk 7]
```diff
@@ -0,0 +1,332 @@
+const chai = require('chai');
+const url = require('url');
+const process = require('process');
```
[the remark this answers is not in the document]
Comment (minsulee2) [reply integration/p2p.test.js]: Yeah, Right I thought this is different one but it isn't! I removed it ! thanks

File: package.json (hunk 2 above)
Comment (minsulee2) [reply package.json]: Removed!

File: p2p/util.js (hunk 3 above)
Comment (minsulee2) [reply p2p/util.js]: Done!

File: p2p/util.js (hunk 4 above)
Comment (minsulee2) [reply p2p/util.js:44]: Done!

File: p2p/util.js (hunk 5 above)
Comment (minsulee2) [reply p2p/util.js]: Done!

File: yarn.lock [hunk 8]
```diff
@@ -3623,6 +3623,11 @@ process-nextick-args@~2.0.0:
   resolved "https://registry.yarnpkg.com/process-nextick-args/-/process-nextick-args-2.0.1.tgz#7820d9b16120cc55ca9ae7792680ae7dba6d7fe2"
   integrity sha512-3ouUOpQhtgrbOa17J7+uxOTpITYWaGP7/AhoR3+A+/1e9skrzelGi/dXzEYyvbxubEF6Wn2ypscTKiKJFFn1ag==
 
+process@^0.11.10:
```
Comment (platfowner) [inline yarn.lock:3626]: can be reverted as well?

File: integration/p2p.test.js (hunk 6 above)
Comment (minsulee2) [reply integration/p2p.test.js]: Agreeeeeeeeed! I thought the same way when I did make the test in the integration test. I will make the integration test with the nodes and for data mock test in the next PR. Thanks!!!

Comment (minsulee2) [reply integration/p2p.test.js]: Done!

**Comment by minsulee2** [comment]
PTAL. All addressed!

**Review by platfowner — APPROVED** [review]
LGTM
Thanks for the nice refactoring and test code!

**Comment by minsulee2** [comment]
Thanks for the fast review!

**Pull request merged**